src/codegen/encoding/scale_encoding.rs

```diff
@@ -578,4 +580,40 @@ impl AbiEncoding for ScaleEncoding {
     fn is_packed(&self) -> bool {
         self.packed_encoder
     }
+
+    fn const_encode(&self, args: &[Expression]) -> Option<Vec<u8>> {
+        let mut result = vec![];
+        for arg in args {
+            match arg {
+                Expression::AllocDynamicBytes {
+                    initializer: Some(data),
+                    ty: Type::String,
```

Comment:
`Type::String | Type::DynamicBytes`

Reply:
Are `Type::DynamicBytes` always string? This will SCALE encode it using `String`, and not just yield the bytes as is

Reply:
`Vec<u8>` and `String` are encoded the  same way in SCALE.

Reply:
While that's true, parsing the vec into an UTF8 string might panic. I think I can add `Type::DynamicBytes` but will add another match arm for it :-)